src/psyclone/psyir/nodes/node.py

```diff
@@ -1149,6 +1149,40 @@ def scope(self):
             "Unable to find the scope of node '{0}' as none of its ancestors "
             "are Container or Schedule nodes.".format(self))
 
+    def replace_with(self, node):
+        '''Replaces self with the supplied node in the PSyIR tree.
+
+        :param node: the node that will replace self in the PSyIR \
+            tree.
+        :type node: subclass of :py:class:`psyclone.psyir.nodes.node`
```

Comment:
I don't know if there is a good reason. Its probably a docstring formatting decision, but in most of the code when we specify a generic concept (a least for psyir.nodes.node this is used a lot) we usually don't have the "subclass of" prepended as it is implicitly understood.

Reply:
Recently we seem to have moved towards using "subclass of" if we are returning an unknown subclass or a base class, although there is no explicit rule. @arporter, do you have any thoughts on this - in general, not for this particular PR.

Reply:
I think Rupert and I in a recent PR agreed to use "subclass of" but, TBH, since a subclass of a class *is* an instance of that class we could drop it for the sake of brevity. I could be persuaded either way although less typing is always good :-)

Reply:
I've removed "subclass of" in this case as requested.